.github/workflows/clang-format-check.yml

```diff
@@ -4,16 +4,29 @@ name: ClangFormat Check
 on:
   push:
     branches:
-      - grass*
+      - grass[0-9]+
+      - master
   pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress:
+    # Do not cancel on protected branches, like grass8
+    ${{ github.ref_protected != true }}
+
+permissions: {}
+
 jobs:
   formatting-check:
     name: Formatting Check
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
```

Comment:
In other cases, you don't explicitly specify `contents: read`. I don't know what is right.

Reply:
I could try removing it to see if it fails. I explicitly flush (remove) all implicit permissions for the complete workflow above on line 18 with 
```
permissions: {}
```
(The 4th example of https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes)

Then I add only for the job what it needs.


For usability reasons (I think), they did not restrict these permissions when they implemented them (after workflows started to be used). One of the goals to achieve is to at least match all the suggestions mentionned in the GitHub Actions Security Hardening articles and blog posts, like this one https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

Reply:
I'm fine with one way or the other. What caught my eye was that it didn't seem consistent.